Make regression tests use default authentication

adutra opened the pull request:
... and align realm names to use `POLARIS` consistently.

Using default authenticators with real JWT tokens, the regression tests are better aligned with a production-like setup.

adutra commented:
@MonkeyCanCode @flyrain @HonahX with this PR we should get regression tests running with production-like profile. This _should_ supersede #866 and fix #865. Let me know what you think.

HonahX commented on regtests/run_spark_sql.sh:
```diff
@@ -55,7 +55,7 @@ if [ -z "${SPARK_HOME}"]; then
 fi
 
-SPARK_BEARER_TOKEN="${REGTEST_ROOT_BEARER_TOKEN:-principal:root;realm:default-realm}"
+SPARK_BEARER_TOKEN="${REGTEST_ROOT_BEARER_TOKEN:-principal:root;realm:POLARIS}"
```
Since the authenticator is `default` now, do we still need this token default value, `principal:root;realm:POLARIS`. Seems we will need to run the added script in `run.sh` to get a token using root credential first before running `run_spark_sql.sh`.

Additionally, is `run_spark_sql.sh` expected to work without first running `run.sh`? If so, may be we should consider adding code to get a token within this script to make it work out-of-box.

adutra replied:
> do we still need this token default value

Indeed I thought about removing the default values everywhere, but I wasn't sure whether all these scripts are always executed from the `run.sh` entrypoint. Do you think we should remove the default values?

> may be we should consider adding code to get a token within this script to make it work out-of-box.

Good catch! The script is definitely meant to be executed individually. I will add support for default auth in that script as well.

adutra replied:
> I will add support for default auth in that script as well.

Done, let me know what you think!

HonahX replied:
> Done, let me know what you think!

Thanks for adding that! Look great!

> but I wasn't sure whether all these scripts are always executed from the run.sh entrypoint. Do you think we should remove the default values?

My understanding is that the default values will always fail since `default` authenticator reject `principal:root:realm:*` token anyway. We will need to either execute `run.sh` to store a real token in env or adding the default auth like `run_spark_sql.sh` to make scripts able to be executed without `run.sh`. 

Keeping default values there will have no affect on tests but I feel it will cause additional work if we want to change realm name for regression test in the future.

adutra replied:
I think I agree. Should I remove the default values in this PR or as a follow-up?

HonahX replied:
I am ok with either way. As you pointed out in https://github.com/apache/polaris/issues/865#issuecomment-2612059626, we will need some follow-up PRs to clean-up/fix things so I think that could also be a good place to remove these values

adutra replied:
Okay, I took note to open a follow-up PR for that 👍

HonahX commented on regtests/docker-compose.yml:
```diff
@@ -31,14 +31,8 @@ services:
-      polaris.realm-context.realms: default-realm,realm1
-      quarkus.log.file.enable: false
+      POLARIS_BOOTSTRAP_CREDENTIALS: POLARIS,root,secret
+      quarkus.log.file.enable: "false"
```
This can also help simplify the docker-compose.yml in `getting-started` examples: e.g. https://github.com/apache/polaris/pull/868

I am happy to update that PR if this one goes in first : )

HonahX reviewed (commented):
@adutra Thanks for updating these!

HonahX reviewed (approved):
LGTM! Thanks for updating these

flyrain commented on README.md:
````diff
@@ -102,16 +99,11 @@ select * from db1.table1;
-```
-
-Then, you can run the regression tests using the following command:
+To run regression tests locally, you first need to start Polaris, with default realm `POLARIS` and
+root credentials: `root:secret`, then run the tests:
````
I guess we don't have to mention this as this code enforces it already.
```
tasks.named<QuarkusRun>("quarkusRun") {
  jvmArgs = listOf("-Dpolaris.bootstrap.credentials=POLARIS,root,secret")
}
```

adutra replied:
You are right, I removed the unnecessary part.

flyrain commented on regtests/run_spark_sql.sh:
```diff
@@ -55,7 +55,21 @@ if [ -z "${SPARK_HOME}"]; then
+if [ "SPARK_BEARER_TOKEN" == "unauthorized_client" ]; then
+  logred "Error: Failed to retrieve bearer token"
+  exit 1
+fi
```
+1, verified it works, thanks for adding it.

flyrain commented on regtests/t_spark_sql/src/spark_sql_s3_cross_region.sh:
```diff
@@ -24,7 +24,7 @@ if [ -z "$AWS_CROSS_REGION_TEST_ENABLED" ] || [ "$AWS_CROSS_REGION_TEST_ENABLED"
 fi
 
-SPARK_BEARER_TOKEN="${REGTEST_ROOT_BEARER_TOKEN:-principal:root;realm:realm1}"
+SPARK_BEARER_TOKEN="${REGTEST_ROOT_BEARER_TOKEN:-principal:root;realm:POLARIS}"
```
Do we need to keep `-principal:root;realm:POLARIS` as token will be fetched by `run.sh`? Given the direction that we want to to is to use the default authenticator always for reggresstion test, I think we should remove them to avoid confusion.

adutra replied:
With @HonahX we decided to postpone that, but I think you are right: leaving this default value will create confusion. So I went ahead and removed them.

flyrain commented on regtests/t_cli/src/test_cli.py:
```diff
@@ -38,7 +38,7 @@ def get_salt(length=8) -> str:
 
 def root_cli(*args):
-    return cli('principal:root;realm:default-realm')(*args)
+    return cli(os.getenv('REGTEST_ROOT_BEARER_TOKEN', 'principal:root;realm:POLARIS'))(*args)
```
Question: Does python regression test also use `run.sh` as I didn't see we add the part to fetch token in python scripts?

flyrain commented on README.md:
```diff
@@ -62,11 +62,8 @@ Apache Polaris is built using Gradle with Java 21+ and Docker 27+.
 - `./gradlew assemble` - To skip tests.
 - `./gradlew test` - To run unit tests and integration tests.
 - `./gradlew polarisServerRun` - To run the Polaris server locally, with profile `prod`; the server 
```
I'm not familiar Quarkus conventions, but "with profile `prod`" seems a bit odd to me, it is no way close to a `prod` deployment. Is "with the default profile" more accurate? We don't even have to mention it in that case.

adutra replied:
You are right. Rephrased.

flyrain commented:
I think we should also remove this line, https://github.com/polaris-catalog/polaris/blob/main/getting-started/trino/README.md#L50-L50, as we changed the way it works. cc @HonahX

adutra commented:
Rebased (to get the tox removal commit) and addressed feedback.

I am now able to run the regression tests both locally and with docker-compose. I am also able to run `./regtests/run_spark_sql.sh` and play interactively with a running Polaris server.

flyrain reviewed (approved):
+1

flyrain commented:
Thanks a lot for the fix @adutra ! Thanks for the review @HonahX!
